### ISSUE TITLE
Bandwidth locked at 4 MB/s on fresh install

### ISSUE THREAD

**Issue by unsigned8547** [Chocobozzz/PeerTube#6260]
### Describe the current behavior

Streaming and downloads run at exactly 4 MB/s (approximately 32 Mb/s) on a fresh 6.0.3 installation. Uploads have no such limit. This speed limit is insufficient for high-quality 4K videos. I am able to reproduce this consistently at multiple web hosts and server configurations. This limit is consistent. This does not appear to be related to bandwidth, routing, or system performance.

Server #1: Contabo hex-core VPS with 16GB RAM. Bidirectional network throughput is consistently over 200 Mb/s, observed both via speedtest and via transfers to/from my home PC.

Server #2: Dedicated E3-1240v5 with 32GB RAM and a 1 TB NVMe drive at a different, unrelated web host in a different datacenter in a different region of the US. Bidirectional network throughput is consistently over 900 Mb/s observed via speedtest, and near or at 300 Mb/s to my home PC (I'm on a 300/300 fiber connection).

I've installed Peertube on both servers using the "any OS" instructions, Cloudron, and Yunohost. A fresh OS install (Debian 11 for "any OS" and Yunohost, Ubuntu 22.04 LTS for Cloudron) was performed between each attempt. After each reinstall, the 4 MB/s limit was observed again. I am able to upload/download at full speed to other apps on the server, including via SFTP. I am able to reproduce the results on both my desktop and laptop. 

As a last ditch effort I used ProtonVPN (using multiple endpoints around the US) from my desktop PC to circumvent any traffic shaping occurring at my ISP or faults with my home LAN. Using a VPN generally only degrades network performance due to increased latency, but even in the tunnel I have more than enough bandwidth to test. Again, downloads and streams were capped at exactly 4 MB/s.

Background:

- 300/300 home ISP connection
- Bidirectional speeds of 200+ (VPS) or 900+ (dedicated server), observed via speedtest, SFTP, and other file apps
- Identical behavior observed firsthand on home ISP (Verizon FIOS) and cellular 5G (T-Mobile)
- Identical behavior observed on my server by third parties at unknown locations
- Able to download at significantly faster speeds from other Peertube servers; does not appear to involve ISP traffic shaping or bandwidth management
- Per Peertube recommendations shown on server options page, web videos are not used; only HLS. I did try adding web videos, and the download rate for them appears to be capped around 3.4 MB/s.

### Steps to reproduce

1. Install fresh OS
2. Ensure OS is up to date (I use apt update / apt upgrade, or followed Cloudron's / Yunohost's recommendations)
3. Install Peertube on either the primary domain or a subdomain; behavior is identical either way
4. Disable web videos, leaving only HLS videos (optional; however, enabling web videos does not seem to have an impact)
5. Select resolutions (I'm using 2160, 1440, 1080, and 720)
6. Create user account
7. Upload video under user account. The videos I'm uploading are all 1.5 - 5 minutes long and in 4K. The longest video is about 6 GB before transcoding.
8. Wait for video to finish transcoding
9. Attempt to stream video. **Observe video pausing as it refills the buffer** 
10. Stop stream
11. Attempt to download video. **Observe download speed maxes out at 4 MB/s (approx 32 Mb/s)**

### Describe the expected behavior

Videos should be able to stream without interruption given sufficient bandwidth between the server and end user. For example, a video with a bitrate of 50 Mb/s should stream perfectly given 300/300 bandwidth.

Likewise, videos should be able to be downloaded as fast as the bandwidth will allow, rather than being limited to 4 MB/s.

### Additional information

* PeerTube instance: 
  * URL: https://video-staging.railtrail.net
  * Version: 6.0.3
  * NodeJS version: 20.11.1
  * Ffmpeg version: 4.3.6-0+deb11u1

* Browser name, version and platforms on which you could reproduce the bug: Brave 1.63.169, Firefox 123.01, Edge 122.0.2365.80

**Comment by SimplyCorbett** [comment]
You have followed the peertube guide and setup the sysctl file with BBR and fq_codel correct?

Anyway little something I learned when trying to stream 5k videos (yes 5k on a modified peertube).

Open the peertube.conf nginx file.

https://github.com/Chocobozzz/PeerTube/blob/develop/support/nginx/peertube

Modify line 216, change from 5M to 7M or your desired speed per client.

Also modify line 223 and 225.

**Comment by unsigned8547** [comment]
Thanks -- I'll check that out as soon as I get to the hotel. Sounds like that's got to be a factor. I'll post an update either way within 24 hours.

**Comment by unsigned8547** [comment]
Thanks for pointing me in the right direction! My hotel has horrible wifi but according to a few beta testers I recruited, videos are now downloading at much more reasonable speeds. I did no further TCP tuning but changing the various limit_rate directives worked. 

Side note to any devs who see this: I'd suggest that pointing this specific directive out in the installation document would be helpful. Alternatively, setting the default to a more 4K-friendly value (like 10M or higher) would help Peertube work smoothly out of the box. I understand the value in facilitating a lean environment, but a 5M limit is barely adequate for 4K 30 FPS and insufficient for 4K 60 FPS. YouTube, for example, recommends more than double that.

Thanks again! Not a bug, but a "could be documented better" directive.
